src/java.base/share/classes/jdk/internal/value/ValueClass.java

```diff
@@ -27,17 +27,30 @@
 
 import jdk.internal.access.JavaLangReflectAccess;
 import jdk.internal.access.SharedSecrets;
+import jdk.internal.misc.PreviewFeatures;
 import jdk.internal.vm.annotation.IntrinsicCandidate;
 
-import java.lang.reflect.Array;
 import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 
 /**
  * Utilities to access package private methods of java.lang.Class and related reflection classes.
  */
-public class ValueClass {
+public final class ValueClass {
     private static final JavaLangReflectAccess JLRA = SharedSecrets.getJavaLangReflectAccess();
 
+    /// {@return whether this field type may store value objects}
+    /// This excludes primitives and includes Object.
+    public static boolean isValueObjectCompatible(Class<?> fieldType) {
+        return PreviewFeatures.isEnabled() && !fieldType.isPrimitive() && (!fieldType.isIdentity() || fieldType == Object.class);
+    }
+
+    /// {@return whether an object of this exact class is a value object}
+    /// This excludes abstract value classes and primitives.
+    public static boolean isValueObjectInstance(Class<?> clazz) {
```

Comment:
The method name is a bit misleading, it seems to ask if the parameter is an instance, but it is a class.
Perhaps `isConcreteValueObject()` or `isConcreteValueClass()`.